commit 16bc119b6b6410863c50d0dafed8c04e5cb6122a
Author: Eran Harary <eran.harary@intel.com>
Date:   Tue Mar 3 13:53:28 2015 +0200

    iwlwifi: trans: Take ownership on secure machine before FW load
    
    When we load the firmware for the 8000 B step device, it'll
    verify its signature. In the current version of the
    hardware, there can be a race between the WiFi firmware
    being loaded and the Bluetooth firmware being loaded.
    
    Check that WiFi is authenticated, if not, take ownership
    on the authentication machine to make sure that the WiFi
    firmware will be authenticated.
    
    Signed-off-by: Eran Harary <eran.harary@intel.com>
    Signed-off-by: Emmanuel Grumbach <emmanuel.grumbach@intel.com>

diff --git a/drivers/net/wireless/iwlwifi/iwl-prph.h b/drivers/net/wireless/iwlwifi/iwl-prph.h
index aa6fb584a5a6..bc962888c583 100644
--- a/drivers/net/wireless/iwlwifi/iwl-prph.h
+++ b/drivers/net/wireless/iwlwifi/iwl-prph.h
@@ -381,6 +381,7 @@ enum secure_load_status_reg {
 #define WFPM_CTRL_REG			0xA03030
 enum {
 	ENABLE_WFPM = BIT(31),
+	WFPM_AUX_CTL_AUX_IF_MAC_OWNER_MSK	= 0x80000000,
 };
 
 #define AUX_MISC_REG			0xA200B0
@@ -388,6 +389,15 @@ enum {
 	HW_STEP_LOCATION_BITS = 24,
 };
 
+#define AUX_MISC_MASTER1_EN		0xA20818
+enum aux_misc_master1_en {
+	AUX_MISC_MASTER1_EN_SBE_MSK	= 0x1,
+};
+
+#define AUX_MISC_MASTER1_SMPHR_STATUS	0xA20800
+#define RSA_ENABLE			0xA24B08
+#define PREG_AUX_BUS_WPROT_0		0xA04CC0
+
 /* FW chicken bits */
 #define LMPM_CHICK			0xA01FF8
 enum {